main_board/src/system/tracing_user.c

```diff
@@ -0,0 +1,25 @@
+#include <compilers.h>
+#include <zephyr/tracing/tracing.h>
+
+#if CONFIG_TRACING_USER == 0
```

Comment:
I think it should rather use `#ifndef`, as this Kconfig value will be undefined when set to `n`, see e.g.
https://github.com/zephyrproject-rtos/zephyr/blob/ef6f804d8f2648935afadaab09b511913ffd9f4c/include/zephyr/tracing/tracing.h#L19
or
https://interrupt.memfault.com/blog/practical_zephyr_kconfig#using-the-applications-kconfig-symbol
```suggestion
#ifndef CONFIG_TRACING_USER
```